### ISSUE TITLE
Add an option to use Huggingface metrics

### ISSUE THREAD

**Issue by yuvalkirstain** [PyTorchLightning/lightning-transformers#206]
## 🚀 Feature

Support Huggingface metrics.

### Motivation

Torchmetrics are great, but there are many metrics that are not publicly available. Luckily, Huggingface implemented lots of them. Can you please an easy way to add metrics from Huggingface?

### Pitch

Specifying a metric from Huggingface, making sure I give it the correct arguments without needing to implement it on my own.

**Comment by mathemusician** [comment]
@yuvalkirstain I think this is a great idea! But after adding a metric to torchmetrics ([link](https://github.com/PyTorchLightning/metrics/pull/668)), I'll be the first to say that this might be a little tricky. There's a lot refactoring going on.

**Comment by mathemusician** [comment]
How did we want to swap metrics? I'm imagining that in the yaml, it looks like:

```yaml
metrics:
  accuracy:
    path: torchmetrics.Accuracy
  recall:
    path: torchmetrics.Recall
```

And we can swap it by switching it to:

```yaml
metrics:
  accuracy:
    path: datasets.metrics.Accuracy
  recall:
    path: datasets.metrics.Recall
```

If there's a custom metric, we can format it like:

```yaml
metrics:
  MyCustomMetric:
    path: abs/path/to/metric.py
```

Where it tries to import MyCustomMetrics from `abs/path/to/metric.py`

Also, with a few metrics, the targets/preds order are switched or are in the process of being switched. So it might be good to generalize metric as a dataclass, just like what HuggingFace is doing, where the output is a callable dictionary. All thoughts on this are welcome!

**Comment by justusschock** [comment]
@mathemusician with only looking at the huggingface metrics API, I think it should be relatively easy to give an adapter for that like this:

```python
class HFMetricAdapter(torchmetrics.Metric):
    def __init__(self, hf_metric):
        super().__init__()
        self.hf_metric = hf_metric

    def update(self, *args, **kwargs):
        self.hf_metric.add_batch(*args, **kwargs)
    
    def compute(self):
        return self.hf_metric.compute()
```

That being said, I don't know much about huggingface metrics and therefore I am not sure, whether all the magic `torchmetrics` does in the background to properly sync and accumulate the metrics state across different GPUs (which would be required for correctness when training on multiple GPUs) will work. For that to work, `huggingface` would have to take care of that.
 
@yuvalkirstain @mathemusician 
What Metrics specifically would you like to see? Maybe we can add more of them to `torchmetrics` for a more native experience?

**Comment by mathemusician** [comment]
@justusschock Good point. It might actually be easier to migrate as many metrics from HuggingFace, then use our hooks, rather writing workarounds for using someone else's hooks. @yuvalkirstain Any thoughts?

**Comment by yuvalkirstain** [comment]
@mathemusician @justusschock 
[Huggingface metrics](https://huggingface.co/metrics) is very popular and might have metrics that can not be found in TorchMetrics, or implementations that slightly differ from those found in TorchMetrics. Additionally, people will probably add new metrics to Huggingface metrics. Therefore, I really like the `HFMetricAdapter` workaround. I find this to be an orthogonal effort to adding more metrics to TorchMetrics.

**Comment by justusschock** [comment]
@yuvalkirstain Understood, as stated, with this adaptor we, however, cannot guarantee that everything works fine regarding synchronization.

And if we start implementing new torchmetrics Metrics now, I think there is a high chance that new metrics can be implemented to it quickly as well. I see a huge benefit there, because this is not tied to one ecosystem.

**Comment by yuvalkirstain** [comment]
@justusschock Yes, I understand. Thank you so much for the help.
